pennylane/gradients/parameter_shift.py

```diff
@@ -327,28 +380,184 @@ def processing_fn(results):
+    return pdA2
+
+
+def _create_variance_proc_fn(
+    tape, var_mask, var_idx, pdA_fn, pdA2_fn, tape_boundary, non_involutory
+):
+    """Auxiliary function to define the processing function for computing the
+    derivative of variances using the parameter-shift rule.
+
+    Args:
+
+        var_mask (list): The mask of variance measurements in the measurement queue.
+        var_idx (list): The indices of variance measurements in the measurement queue.
+        pdA_fn (callable): The function required to evaluate the analytic derivative of <A>.
+        pdA2_fn (callable): If not None, non-involutory observables are
+            present; the partial derivative of <A^2> may be non-zero. Here, we
+            calculate the analytic derivatives of the <A^2> observables.
+        tape_boundary (callable): the number of first derivative tapes used to
+            determine the number of results to post-process later
+        non_involutory (list): the indices in the measurement queue of all non-involutory
+            observables
+    """
+
+    def processing_fn(results):
+        # HOTFIX: Apply the same squeezing as in qml.QNode to make the transform output consistent.
+        # pylint: disable=protected-access
+        if (
+            not qml.active_return()
+            and tape._qfunc_output is not None
+            and not isinstance(tape._qfunc_output, Sequence)
+        ):
+            results = [qml.math.squeeze(res) for res in results]
+
+        # We need to expand the dimensions of the variance mask,
+        # and convert it to be the same type as the results.
+        res = results[0]
+
+        mask = []
+        for m, r in zip(var_mask, qml.math.atleast_1d(results[0])):
+            array_func = np.ones if m else np.zeros
+            shape = qml.math.shape(r)
+            mask.append(array_func(shape, dtype=bool))
+
+        ragged = getattr(results[0], "dtype", None) is np.dtype("object")
+        if not qml.active_return() and ragged and qml.math.ndim(res) > 0:
+            res = qml.math.hstack(res)
+            mask = qml.math.hstack(mask)
+
+        # Note: len(f0) == len(mask)
+        f0 = qml.math.expand_dims(res, -1)
+        mask = qml.math.convert_like(qml.math.reshape(mask, qml.math.shape(f0)), res)
+
+        pdA = pdA_fn(results[1:tape_boundary])
+
+        pdA2 = _get_pda2(results, tape, pdA2_fn, tape_boundary, non_involutory, var_idx)
+
+        # return d(var(A))/dp = d<A^2>/dp -2 * <A> * d<A>/dp for the variances (mask==True)
+        # d<A>/dp for plain expectations (mask==False)
+
+        # Note: if isinstance(pdA2, int) and pdA2 != 0, then len(pdA2) == len(pdA)
+
+        if qml.active_return():
+            multi_measure = len(tape.measurements) > 1
+            if multi_measure:
+
+                res = []
+                for idx, m in enumerate(mask):
+
+                    if isinstance(pdA2, np.ndarray) and isinstance(pdA2, (Sequence, np.ndarray)):

```

Comment:
Is there a typo? The first condition is included in the second.

Reply:
Yes! Thanks for the catch.